.github/workflows/server-benchmark.yml

```diff
@@ -24,17 +26,102 @@ jobs:
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
         restore-keys: |
           ${{ runner.os }}-gradle-
-    # JDK is needed to generate code
+      # Pinned to the commit hash of v1.3.0
+    - uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
     - name: Set up JDK
       uses: actions/setup-java@v1
       with:
         java-version: ${{ env.java_version }}
-    # Install Rust
-    - uses: actions-rs/toolchain@v1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
       with:
         toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
         default: true
     - name: Run integration tests
       run: |
         cd rust-runtime/aws-smithy-http-server/examples && \
           make && cargo test
+
+  run-benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout PR
+      uses: actions/checkout@v3
+      with:
+        path: pull-request
+    - name: Checkout origin/main
+      uses: actions/checkout@v3
+      with:
+        repository: awslabs/smithy-rs
+        path: origin-main
+        # ref: main
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I will change this back to use `ref: main` just before pushing, otherwise we are not able to run the benchmark since it is not yet in `main`.